### PR TITLE
Fix the way the pending requests module is used in settled.ts

### LINKED ISSUE #1172: EmberApplicationInstance willDestroy in settled.ts uses the wrong `internalPendingRequests`
upon updating ember-test-helpers, I have test code that resets the app hit the app instance `willDestroy` method, which errors on `internalPendingRequests.clearPendingRequests()` because `internalPendingRequests` is the (nonzero) integer result of calling the `pendingRequests` method.

Here's where the pendingRequests method is chained on the require that prevents accessing the other methods:
https://github.com/emberjs/ember-test-helpers/blob/9c6e884d8572e7c262e4915d88dd61df787deda9/addon-test-support/%40ember/test-helpers/settled.ts#L30-L31

error occurs here:
https://github.com/emberjs/ember-test-helpers/blob/9c6e884d8572e7c262e4915d88dd61df787deda9/addon-test-support/%40ember/test-helpers/settled.ts#L59

likewise the `off` calls are passing `undefined` since `internalPendingRequests` is an integer and not a module

If you look at the deleted code in https://github.com/emberjs/ember-test-helpers/pull/1047 it was not chaining `pendingRequests` on the old `require('ember-testing/lib/test/pending_requests')` which allowed accessing the different methods from that module.

Is a PR welcome here?

### PR DESCRIPTION
**Pull request by raycohen**
Fixes #1172 
Related to #1047

Before these changes, the code is confusing a module with one of the functions on that module.

### CODE REVIEW

File: addon-test-support/@ember/test-helpers/settled.ts [hunk 1]
```diff
@@ -44,20 +49,16 @@ if (typeof jQuery !== 'undefined' && _internalPendingRequests) {
   // This can be removed once Ember 4.0.0 is released
   EmberApplicationInstance.reopen({
     willDestroy(...args: any[]) {
-      const internalPendingRequests = _internalPendingRequests();
-
-      if (internalPendingRequests !== 0) {
```
Comment (raycohen) [inline addon-test-support/@ember/test-helpers/settled.ts:49]: the outer if statement checks that `_internalPendingRequestsModule` exists, which I believe makes calling `off` and `clearPendingRequests` safe and desirable regardless of what the PendingRequestsCount is

**Review by rwjblue — APPROVED** [review]
Thanks @raycohen!

**Comment by ginomiglio** [comment]
Would it be possible to cut a release that includes this change?

**Comment by raycohen** [comment]
@rwjblue any chance we can get a release with this 🙏

**Comment by rwjblue** [comment]
oh dear, sorry y'all; I thought I had already released it 😭

**Comment by rwjblue** [comment]
Finally released in 2.6.2

**Pull request merged**